schema.sql

```diff
@@ -1,29 +1,53 @@
 CREATE TABLE animals (
-    id SERIAL PRIMARY KEY, name VARCHAR(255), date_of_birth DATE,
-    escape_attempts INTEGER, neutered BOOLEAN, weight_kg DECIMAL
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255),
+  date_of_birth DATE,
+  escape_attempts INTEGER,
+  neutered BOOLEAN,
+  weight_kg DECIMAL
 );
 
 ALTER TABLE animals ADD COLUMN species VARCHAR(255);
 
-CREATE TABLE owners (id SERIAL PRIMARY KEY, full_name VARCHAR(255), age INTEGER);
-CREATE TABLE species(id SERIAL PRIMARY KEY, name VARCHAR(255));
+CREATE TABLE owners (
+  id SERIAL PRIMARY KEY,
+  full_name VARCHAR(255),
+  age INTEGER
+);
+
+CREATE TABLE species(
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255)
+);
 
 ALTER TABLE animals DROP COLUMN species;
 ALTER TABLE animals ADD COLUMN species_id INTEGER REFERENCES species(id);
 ALTER TABLE animals ADD COLUMN owner_id INTEGER REFERENCES owners(id);
 
-CREATE TABLE vets(id SERIAL PRIMARY KEY, name VARCHAR(255), age INTEGER, date_of_graduation DATE);
+CREATE TABLE vets(
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255),
+  age INTEGER,
+  date_of_graduation DATE
+);
 
 CREATE TABLE visits(animal_id INTEGER,
   vet_id INTEGER, 
   PRIMARY KEY(animal_id, vet_id),
   visit_date Date,
   FOREIGN KEY (animal_id) REFERENCES animals (id)
-  FOREIGN KEY (vet_id) REFERENCES vets (id));
+  FOREIGN KEY (vet_id) REFERENCES vets (id)
+);
 
 CREATE TABLE specializations(vet_id INTEGER,
   species_id INTEGER,
   PRIMARY KEY(vet_id, species_id), 
   FOREIGN KEY(vet_id) REFERENCES vets (id),
-  FOREIGN KEY (species_id) REFERENCES species (id));
+  FOREIGN KEY (species_id) REFERENCES species (id)
+);
+
+ALTER TABLE owners ADD COLUMN email VARCHAR(120);
 
+CREATE INDEX index_visits_animal_id ON visits(animal_id);
+CREATE INDEX idx_visits_vet_id ON visits(vet_id);
```

Comment:
Great job on adding indexes 🤝
- However for this query `SELECT * FROM visits where vet_id = 2;` and according to the results in the output provided. There seems to be no reduction in the execution time but instead, the execution time increased. I would kindly recommend you remove the indexes and add them again to see if the execution time reduces.
- Before
![before](https://github.com/Esmatullah12/vet-clinic-database/assets/63917725/28d0b584-c1bc-4675-b553-2f76d0ddbceb)
- After
![after](https://github.com/Esmatullah12/vet-clinic-database/assets/63917725/b7ebd6f7-78e0-4db1-95d4-d2b2fefa90ce)

Reply:
Remove the existing indexes from the visits table, and then reapply them to enhance performance. The screenshot below shows the results after removing and re-adding the indexes.
<img width="864" alt="Screenshot 2023-11-08 at 09 53 45" src="https://github.com/Esmatullah12/vet-clinic-database/assets/101590645/57ac9106-abc2-4f01-851d-c486c4bec110">